lib/model/query/user-preferences.js

```diff
@@ -0,0 +1,34 @@
+// Copyright 2024 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const { sql } = require('slonik');
+
+const getForUser = (userId) => ({ maybeOne }) =>
+  maybeOne(sql`select body from userpreferences where "userId"=${userId}`);
+
+const setForUser = (userId, body) => ({ one }) => {
+  const json = body != null ? JSON.stringify(body) : null;
+  return one(sql`
+    insert into userpreferences ("userId", body) values (${userId}, ${json})
+    on conflict ("userId") do update set body=${json}
+    returning 1
+  `);
+};
```

Comment:
Related to some of my comments below about frontend probably not differentiating between creating a new set of preferences or updating an existing set, maybe we only need one type of method here that can also handle: 

- change existing preference
- add new preference to existing set
- add new preference to empty set

Reply:
Oh, so you're thinking this part of user prefs would look like `formTrashCollapse: [projId1, projId7, projId9]`? That wasn't obvious to me at first.

I feel like we should pick a structure that will work for future project-specific settings like the ones in the issue about form sort order and booleans about showing legacy QR codes. Something like `projId: {collapseTrash: true, formSort: 'latest', showLegacyQRcodes: true}`